pdns/syncres.cc

```diff
@@ -3590,13 +3590,25 @@ bool SyncRes::processRecords(const std::string& prefix, const DNSName& qname, co
       if (moreSpecificThan(rec.d_name,auth)) {
         newauth = rec.d_name;
         LOG(prefix<<qname<<": got NS record '"<<rec.d_name<<"' -> '"<<rec.d_content->getZoneRepresentation()<<"'"<<endl);
-        realreferral = true;
+
+        if (!negindic && qtype == QType::DS && qname == newauth) {
+          /* just got a referral from the parent zone when asking for a DS, looks like this server did not get the DNSSE memo.. */
```

Comment:
Typo in DNSSEC. Maybe we want to check if RRSIG's/NSEC's are included in the answer? Cause that would mean it does support DNSSEC but doesn't properly respond to a DS records

Reply:
Thanks! I spotted the typo roughly at the same time you did, apparently :)

For the RRSIG/NSEC check, if we encounter a NSEC(3) denying the DS we will override that decision (and not even enter that block because of the `negindic` check if we have already seen the NSEC(3)). I'll see if I can make that cleaner, and add the very least add a comment or two.

Reply:
Got a cleaner code and a test case to make sure we don't mess that up in Secure zones.